src/gsi/read_gnssrspd.f90

```diff
@@ -0,0 +1,514 @@
+     real(r_kind) :: obstime(6,1)
+     real(r_kind) :: obsloc(2,1)
+     real(r_kind) :: obstmp(2,1)
+     real(r_kind) :: obswnd(4,1)
+     real(r_kind) :: obsfmr(2,1)
+     real(r_kind) :: obsmst(3,1)
+     real(r_kind) :: obsprs(1,1)
+     real(r_kind) :: obspsf(1,1)
+     real(r_kind) :: obsg10(1,1)
+     real(r_kind) :: obsqcm(2,1)
+     real(r_kind) :: gnssrw(2,1)
+      
+     real(r_double) :: rstation_id
+     real(r_double) :: r_prvstg(1,1),r_sprvstg(1,1)
+
+     real(r_kind), allocatable,dimension(:,:) :: cdata_all,cdata_out
+     real(r_kind), allocatable,dimension(:)   :: presl_thin
+
+!    Equivalence to handle character names
+     equivalence(r_prvstg(1,1),c_prvstg)
+     equivalence(r_sprvstg(1,1),c_sprvstg)
+     equivalence(rstation_id,c_station_id)
+
+!    Data 
+     data hdstr / 'SID' /
+     !data timestr  / 'YEAR MNTH DAYS HOUR MINU SECO' /
+     data timestr / 'DHR RPT' /
+     !data locstr   / 'CLAT CLON' /
+     data locstr   / 'XOB YOB' /
+     !data wndstr   / 'WSPD' / !GNSSRSPD Wind speed
+     data wndstr   / 'SOB' / !GNSSRSPD Wind speed
+     data oestr   / 'WSU' / !GNSSRSPD Wind speed uncertainty/error 
+     data lunin    / 13 /
+     data ithin    / -9 /
+     data rmesh    / -99.999_r_kind /
+ 
+!------------------------------------------------------------------------------------------------
+
+     write(6,*)'READ_GNSSRSPD: begin to read gnssrspd satellite data ...'
+
+!    Initialize parameters
+
+!    Set common variables
+     lspdob = obstype == 'gnssrspd'
+
+     nreal  = 0
+     iecol  = 0
+ 
+ 
+     lim_qm = 4
+     iecol=0
+     if (lspdob) then
+        nreal  = 23
+        iecol  =  4  
+        errmin = one
+     else 
+        write(6,*) ' illegal obs type in read_gnssrspd '
+        call stop2(94)
+     end if
+
+     inflate_error = .true.
+
+!    Check if the obs type specified in the convinfo is in the fl hdob bufr file 
+!    If found, get the index (nc) from the convinfo for the specified type
+     ntmatch =  0
+     ncsave  =  0
+     do nc = 1, nconvtype
+               if (trim(ioctype(nc)) == trim(obstype)) then 
+                  if (trim(ioctype(nc)) == 'gnssrspd' .and. ictype(nc) == 298 ) then
+               ntmatch = ntmatch+1
+               ncsave  = nc
+               itype   = ictype(nc)
+           end if
+        end if
+     enddo
+     if(ntmatch == 0)then  ! Return if not specified in convinfo 
+        write(6,*) ' READ_GNSSRSPD: No matching obstype found in obsinfo ',obstype
+        return
+     else 
+        nc = ncsave
+        write(6,*) ' READ_GNSSRSPD: Processing GNSSRSPD data : ', ntmatch, nc, ioctype(nc), ictype(nc), itype 
+     end if
+
+
+!------------------------------------------------------------------------------------------------
+
+!    Go through the bufr file to find out how mant subsets to process
+     nmsg   = 0
+     maxobs = 0
+     call closbf(lunin) 
+     open(lunin,file=trim(infile),form='unformatted')
+     call openbf(lunin,'IN',lunin)
```

Comment:
@KariA-Spire The Input data is BUFR format.  Do you bufrize the data yourself or it is done by NESDIS?

Reply:
We do bufrize the data.